### PR TITLE
[webui] Add group tasks to user#show

### PR DESCRIPTION
**Pull request by ChrisBr**

### CODE REVIEW

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -1204,10 +1204,15 @@ def update_cache
...
     user_ids << User.find_by_login!(creator).id
+
+    group_ids = Relationship.where(package_id: target_package_ids).or(
+      Relationship.where(project_id: target_project_ids)
+    ).groups.pluck('group_id')
     # rubocop:disable Rails/SkipsModelValidations
     # Skipping Model validations in this case is fine as we only want to touch
     # the associated user models to invalidate the cache keys
     User.where(id: user_ids).update_all(updated_at: Time.now)
+    Group.where(id: group_ids).update_all(updated_at: Time.now)
```
Comment (evanrolfe) [inline src/api/app/models/bs_request.rb]: [Nitpic] theres no need to do a second where query here, i think you can just do this?
```ruby
Relationship.where(package_id: target_package_ids).or(
  Relationship.where(project_id: target_project_ids)
).groups.update_all(updated_at: Time.now)
```

File: src/api/app/controllers/webui/user_controller.rb [hunk 2]
```diff
@@ -55,6 +55,8 @@ def show
     @iprojects = @displayed_user.involved_projects.pluck(:name, :title)
     @ipackages = @displayed_user.involved_packages.joins(:project).pluck(:name, 'projects.name as pname')
     @owned = @displayed_user.owned_packages
+    @groups = @displayed_user.groups
+    @roles = @displayed_user.roles.global.pluck(:title)
```
Comment (evanrolfe) [inline src/api/app/controllers/webui/user_controller.rb]: [Nitpick] Can you name this `@role_titles` instead?

File: src/api/spec/models/bs_request_spec.rb [hunk 3]
```diff
@@ -116,8 +116,8 @@
   end
 
   describe '#update_cache' do
-    RSpec.shared_examples 'bs_request after_commit callback' do
-      it 'touches the user' do
+    RSpec.shared_examples 'has associated object' do
```
Comment (evanrolfe) [inline src/api/spec/models/bs_request_spec.rb]: [Nitpick] "has associated object" doesn't really make sense to me, it seems like this shared context is more about making sure the users request cache is reset? Maybe say `it_should_behave_like "the user's cache is reset"`?

File: src/api/app/models/bs_request.rb (hunk 1 above)
Comment (ChrisBr) [reply src/api/app/models/bs_request.rb]: Hm, good idea. However, groups is at the moment only a scope of the Relationship model (``where.not(group_id: nil)``). I will investigate.

**Comment by ChrisBr** [comment]
The code climate changes are unrelated, tried to mark them as resolved but seems like I lack permissions

**Pull request merged**